src/Metadata/Resource/Factory/ResourceMetadataFactoryTrait.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace ApiPlatform\Core\Metadata\Resource\Factory;
+
+use ApiPlatform\Core\Exception\ResourceClassNotFoundException;
+use ApiPlatform\Core\Metadata\Resource\Operation;
```

Comment:
This is already working here, I've a working proof but it needs the #498 fix to work:

```
    foobar:
        class: 'Acme\Entity\Foobar'
        collectionOperations: 
            get: 
                method: 'GET'
            post: 
                method: 'POST'
            by_party:
                method: 'GET'
                route: '/foo_bar_by_bar/{foobar}'
                controller: 'acme.action.foobar'
        attributes:
            normalization_context: 
                groups: ['default']
            denormalization_context: 
                groups: ['default']
```